### PR TITLE
Rollup: transform/transformBundle can return a promise

### PR DESCRIPTION
**Pull request by avgaltsev**
I didn't find that in documentation, but this is obvious it's possible to return a promise from transform/transformBundle.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [transform](https://github.com/rollup/rollup/blob/1ef720f551d97f72b2aecea2bfaed3c7a1fb2605/src/utils/transform.ts#L82), [transformBundle](https://github.com/rollup/rollup/blob/1ef720f551d97f72b2aecea2bfaed3c7a1fb2605/src/utils/transformBundle.ts#L18)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by mhegazy** [comment]
@alan-agius4 @plesiecki @flying-sheep can you please review this change.

File: types/rollup/index.d.ts [hunk 1]
```diff
@@ -138,9 +138,9 @@ export interface Plugin {
 	 */
 	resolveId?(importee: string, importer: string | undefined): string | null | undefined | false | 0 | ''
 	/** A module transformer function */
-	transform?(this: TransformContext, source: string, id: string): string | null | undefined | { code: string, map: SourceMap }
+	transform?(this: TransformContext, source: string, id: string): TransformResult | Promise<TransformResult>
```
Comment (alan-agius4) [inline types/rollup/index.d.ts]: These seems to retune only a promise

Comment (alan-agius4) [reply types/rollup/index.d.ts]: Check this as a guidance https://github.com/rollup/rollup/blob/master/src/rollup/index.ts

**Review by alan-agius4 — CHANGES_REQUESTED** [review]

File: types/rollup/index.d.ts (hunk 1 above)
Comment (avgaltsev) [reply types/rollup/index.d.ts]: But there is still a possibility to return instant values from these hooks. Moreover the official [plugin guidance](https://github.com/rollup/rollup/wiki/Plugins#creating-plugins) (maybe outdated) says no word about promises. In [official tests](https://github.com/rollup/rollup/blob/master/test/form/samples/absolute-path-resolver/_config.js) we can see transform hooks without promises.

Comment (alan-agius4) [reply types/rollup/index.d.ts]: That is right, thanks for the clariffication.

**Comment by alan-agius4** [comment]
I approve!

**Comment by alan-agius4** [comment]
https://github.com/rollup/rollup/issues/1842

**Pull request merged**